modules/moderation.py

```diff
@@ -629,126 +545,91 @@ async def mod_error(self, ctx, error):
             await ctx.send(f'{config.redTick} An unknown exception has occured, if this continues to happen contact the developer.', delete_after=15)
             raise error
 
-class LoopTasks(commands.Cog):
-    def __init__(self, bot):
-        self.bot = bot
-        self.NS = self.bot.get_guild(config.nintendoswitch)
-        self.modLogs = self.bot.get_channel(config.modChannel)
-        self.adminChannel = self.bot.get_channel(config.adminChannel)
-        self.expiryWarnNotified = {}
-        self.roles = {
-            'tier1': self.NS.get_role(config.warnTier1),
-            'tier2': self.NS.get_role(config.warnTier2),
-            'tier3': self.NS.get_role(config.warnTier3),
-            'mute': self.NS.get_role(config.mute)
-        }
-        self.expiry_check.start() #pylint: disable=no-member
-        self.expiry_check.add_exception_type(discord.errors.DiscordServerError) #pylint: disable=no-member
-        logging.info('[Cog] Moderation tasks cog loaded')
-
-    def cog_unload(self):
-        logging.info('[Cog] Attempting to stop task expiry_check...')
-        self.expiry_check.stop() #pylint: disable=no-member
-        logging.info('[Cog] Task expiry_check exited')
-        logging.info('[Cog] Moderation tasks cog unloaded')
-
-    @tasks.loop(seconds=30)
-    async def expiry_check(self):
-        logging.debug('[Moderation] Starting expiry check')
+    async def expire_actions(self, _id, guild):
         db = mclient.bowser.puns
-        activePuns = db.find({'active': True, 'expiry': {'$ne': None}})
-        if not activePuns.count():
-            #logging.info('[Moderation] No active puns to cycle through')
+        doc = db.find_one({'_id': _id})
+        if not doc:
+            logging.error(f'[Moderation] Expiry failed. Doc {_id} does not exist!')
             return
-        #print(f'{activePuns.count()}')
 
-        warns = ['tier1', 'tier2', 'tier3']
-        for pun in activePuns:
-            await asyncio.sleep(0.01) # Give some breathing room to the rest of the thread as this is more long running
-            #print('processing pun')
-            try:
-                member = await self.NS.fetch_member(pun['user'])
-
-            except discord.NotFound: # User not in guild
-                continue
+        # Lets do a sanity check.
+        if not doc['active']:
+            logging.debug(f'[Moderation] Expiry failed. Doc {_id} is not active but was scheduled to expire!')
+            return
 
+        twelveHr = 60 * 60 * 12
+        if doc['type'] == 'strike':
+            userDB = mclient.bowser.users
+            user = userDB.find_one({'_id': doc['user']})
             try:
-                moderator = await self.NS.fetch_member(pun['moderator'])
+                if user['strike_check'] > time.time(): # To prevent drift we recall every 12 hours. Schedule for 12hr or expiry time, whichever is sooner. 5 seconds is for drift lienency
```

Comment:
"5 seconds is for drift lienency" comment is unclear what it refers to

Reply:
This is apparently an outdated comment, see https://github.com/rNintendoSwitch/MechaBowser/pull/37#discussion_r554703680 for a similar example.

Reply:
Resolved in 79046a7